### PR TITLE
HOC feature

### PR DESCRIPTION
**Pull request by evenchange4**
feat(HOC): simple implement with recompose that wrap Helmet component into HOC.

resolve [#67](https://github.com/nfl/react-helmet/issues/67)

### CODE REVIEW

File: src/withHelmet.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* eslint react/display-name: 0 */
+import React from "react";
+import createElement from "recompose/createElement";
+import createHelper from "recompose/createHelper";
+import Helmet from "./Helmet.js";
+
+const withHelmet = (propsMapper, BaseComponent) => props => (
+  <div>
```
Comment (mull) [inline src/withHelmet.js]: Curious, what's the reason the Helmet component can't just `return this.props.children`? These empty divs tend to mess things up when `height: 100%` is desired further down.

Comment (cwelch5) [inline src/withHelmet.js]: @mull Helmet doesn't take children as it messes with the `shouldComponentUpdate` check.

**Comment by cwelch5** [comment]
@evenchange4 Thx for the PR.  I haven't worked with `recompose` it looks interesting.  I need to research a bit.

File: src/withHelmet.js (hunk 1 above)
Comment (evenchange4) [inline src/withHelmet.js]: Yes, the empty `div` wrapper may cause side effect in some styling case.
But I have not come up with idea to work with `react-side-effect`.
Any suggestion to improve it ?

**Comment by leesiongchan** [comment]
+1

File: package.json [hunk 2]
```diff
@@ -29,12 +29,13 @@
     "deep-equal": "1.0.1",
     "invariant": "2.2.0",
     "react-side-effect": "1.0.2",
+    "recompose": "^0.14.5",
```
Comment (srph) [inline package.json]: I'm not sure if the tradeoff for a new dependency for the small yet convenient helpers is worth it. It's not that much work anyway to make the HOC for react-helmet.

Comment (cwelch5) [inline package.json]: I would tend to agree.

Comment (cwelch5) [inline package.json]: I agree with @srph and feel like we should accomplish this without adding another dependency to the library.  @evenchange4 - Thank you for your PR, if you'd like to give it another try without adding a dependency, I'll definitely take a look.

**Pull request closed without merging**